### PR TITLE
format cqc location date fields

### PR DESCRIPTION
**Pull request by james-francis-MT**
# Description
Formatting date fields in cqc location data

# How to test
Can be tested in main once merged

[Trello ticket](https://trello.com/c/sL59mLvc/138-epic-4-format-date-fields-must)

# Developer checklist
- [x] Unit test
- [x] Linked to Trello ticket
- [x] Documentation up to date

### CODE REVIEW

File: jobs/clean_cqc_location_data.py [hunk 1]
```diff
@@ -30,6 +30,8 @@
 NONE_NURSING_HOME_IDENTIFIER = "Care home without nursing"
 NONE_RESIDENTIAL_IDENTIFIER = "non-residential"
 
+DATE_COLUMN_IDENTIFIER = "registration_date"
```
Comment (MarionJHolloway) [inline jobs/clean_cqc_location_data.py:33]: There are multiple date columns in this dataset that will want formatting. If you look at the function definition, you can omit this parameter, as is will run on all columns with date in the column name.

File: tests/unit/test_clean_cqc_location_data.py [hunk 2]
```diff
@@ -55,6 +57,7 @@ def test_main_runs(
 
         self.assertEqual(remove_already_cleaned_data_patch.call_count, 1)
         self.assertEqual(read_from_parquet_patch.call_count, 2)
+        format_date_fields_mock.assert_called_once()
```
Comment (MarionJHolloway) [inline tests/unit/test_clean_cqc_location_data.py:60]: Will probably need to update this too - maybe to check the call count?

**Review by MarionJHolloway — APPROVED** [review]
Just a couple of small changes to make before merging.

File: jobs/clean_cqc_location_data.py (hunk 1 above)
Comment (james-francis-MT) [reply jobs/clean_cqc_location_data.py:33]: I had a look at the schema and the only date columns either had this in the name or were 'import_date'. This behaviour is what currently happens in prepare_locations. Is it okay to also attempt to convert import date to a date type?

File: tests/unit/test_clean_cqc_location_data.py (hunk 2 above)
Comment (james-francis-MT) [reply tests/unit/test_clean_cqc_location_data.py:60]: https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.assert_called_once This checks that the function was called exactly once

Comment (MarionJHolloway) [reply tests/unit/test_clean_cqc_location_data.py:60]: Sorry, I was thinking it was called once for each date column - Friday brain!

File: jobs/clean_cqc_location_data.py (hunk 1 above)
Comment (MarionJHolloway) [reply jobs/clean_cqc_location_data.py:33]: It might be worth having a chat about this one because it will impact the date alignment too

**Pull request merged**